test/Table_Tests/src/Util.enso

```diff
@@ -47,6 +32,46 @@ DB_Column.should_equal self expected frames_to_skip=0 =
     t1 = expected.read
     t0 . should_equal t1 frames_to_skip
 
+## PRIVATE
+Table.should_equal_impl self expected loc =
+    case expected of
+        _ : Table ->
+            if self.columns.length != expected.columns.length then
+                Panic.throw 'Tables differ at '+loc+'.\nActual:\n'+self.display+'\nExpected:\n'+expected.display+'\nExpected '+expected.columns.length.to_text+" columns, but got "+self.columns.length.to_text+'.'
+            Panic.catch Any (self.columns.zip expected.columns a-> e->(a.should_equal_impl e)) error->
+                msg = 'Tables differ at '+loc+'.\nActual:\n'+self.display+'\nExpected:\n'+expected.display+'\n'+error.payload
+                Panic.throw msg
+        _ -> Panic.throw "Got a Table, but expected a "+expected.to_display_text+(display_loc loc)+'.'
+
+## PRIVATE
+Column.should_equal_impl self expected loc='' =
+    case expected of
+        _ : Column ->
+            if self.name != expected.name then
+                Panic.throw "Expected column name "+expected.name+", but got "+self.name+(display_loc loc)+'.'
+            if self.length != expected.length then
+                Panic.throw "Expected column length "+expected.length.to_text+", but got "+self.length.to_text+(display_loc loc)+'.'
+            if self.value_type != expected.value_type then
+                Panic.throw "Expected column type "+expected.value_type.to_text+", but got "+self.value_type.to_text+(display_loc loc)+'.'
+            self.zip expected a-> e->
+                if a != e then
+                    if (a.is_a Number && e.is_a Number && a.is_nan && e.is_nan).not then
```

Comment:
This logic is duplicated here and then re-checked again in `report_fail`. Couldn't we merge this into a single check?

Reply:
Done